xlators/cluster/ec/src/ec.c

```diff
@@ -716,61 +716,24 @@ ec_statistics_init(ec_t *ec)
 static int
 ec_assign_read_mask(ec_t *ec, char *read_mask_str)
 {
-    char *mask = NULL;
-    char *maskptr = NULL;
-    char *saveptr = NULL;
-    char *id_str = NULL;
-    int id = 0;
     int ret = 0;
-    uintptr_t read_mask = 0;
+    uintptr_t read_mask;
 
     if (!read_mask_str) {
         ec->read_mask = 0;
         ret = 0;
         goto out;
     }
 
-    mask = gf_strdup(read_mask_str);
-    if (!mask) {
-        ret = -1;
-        goto out;
-    }
-    maskptr = mask;
-
-    for (;;) {
-        id_str = strtok_r(maskptr, ":", &saveptr);
-        if (id_str == NULL)
-            break;
-        if (gf_string2int(id_str, &id)) {
-            gf_msg(ec->xl->name, GF_LOG_ERROR, 0, EC_MSG_XLATOR_INIT_FAIL,
-                   "In read-mask \"%s\" id %s is not a valid integer",
-                   read_mask_str, id_str);
-            ret = -1;
-            goto out;
-        }
-
-        if ((id < 0) || (id >= ec->nodes)) {
-            gf_msg(ec->xl->name, GF_LOG_ERROR, 0, EC_MSG_XLATOR_INIT_FAIL,
-                   "In read-mask \"%s\" id %d is not in range [0 - %d]",
-                   read_mask_str, id, ec->nodes - 1);
-            ret = -1;
-            goto out;
-        }
-        read_mask |= (1UL << id);
-        maskptr = NULL;
-    }
-
-    if (gf_bits_count(read_mask) < ec->fragments) {
-        gf_msg(ec->xl->name, GF_LOG_ERROR, 0, EC_MSG_XLATOR_INIT_FAIL,
-               "read-mask \"%s\" should contain at least %d ids", read_mask_str,
-               ec->fragments);
+    read_mask = ec_parse_read_mask(ec, read_mask_str, EC_MSG_XLATOR_INIT_FAIL);
+    if (read_mask == 0) {
```

Comment:
read_mask being zero is not an error, we have been treating it equivalent to all bricks